django/conf/__init__.py

```diff
@@ -86,8 +86,17 @@ def __getattr__(self, name):
         # This is done here for performance reasons so the modified value is cached.
         if name in {'MEDIA_URL', 'STATIC_URL'} and val is not None:
             val = self._add_script_prefix(val)
-        elif name == 'SECRET_KEY' and not val:
-            raise ImproperlyConfigured("The SECRET_KEY setting must not be empty.")
+
+        # Normalization of SECRET_KEY(S) happens in Settings.__init__(), and
+        # override_settings(). They're validated by system checks. Here we
+        # simply raise on an empty value.
+        if name == 'SECRET_KEY' and not val:
+            raise ImproperlyConfigured('The SECRET_KEY setting must not be empty.')
+        elif name == 'SECRET_KEYS':
+            if not val:
+                raise ImproperlyConfigured('The SECRET_KEYS setting must not be empty.')
+            if not val[0]:
```

Comment:
nit: since the error message says "the elements" shouldn't this check all elements?
```suggestion
            if not all(element for element in val):
```
alternatively, we can leave the if-statement as-is and instead change the error to something like "the first key in the list must not be empty"

Reply:
The system check already does that. This is just a runtime safety net really. We can extend it or adjust the message but I'm not sure we need to handle it too robustly.